qsize.c

```diff
@@ -95,8 +97,19 @@ struct qsize_opt_state {
 
 	size_t num_all_files, num_all_nonfiles, num_all_ignored;
 	uint64_t num_all_bytes;
+
+	set *inodes;
 };
 
+/* Returns whether the dev,inode pair is a new pair that hasn't been seen so far */
+static bool add_inode(struct qsize_opt_state *state, dev_t dev, ino_t inode) {
+	char key[2*sizeof(unsigned long long) + 1 + 2*sizeof(unsigned long long) + 1];
+	assert(sprintf(key, "%llx/%llx", (unsigned long long)dev, (unsigned long long)inode) < (ssize_t)sizeof(key));
+	bool new_pair;
+	state->inodes = add_set_unique(key, state->inodes, &new_pair);
+	return new_pair;
+}
```

Comment:
this function is nothing now, just inline it, which is much cheaper because then the printf buffer is allocated only once

Reply:
No, it would not be cheaper, because:
- The sprintf buffer (variable: key) takes no CPU instructions to allocate because it is a stack variable with a constant compile-time size
- At -O2 optimization level, gcc will automatically inline the static `add_inode` function into its single call-site